cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java

```diff
@@ -775,19 +775,54 @@ private boolean checkEventTriggerCondition(BannerTriggerCondition condition) {
   }
 
   private boolean checkTargetEventRelationFilter(CheckFilterRelation relation, String compareValue,
-                                                 String attributeValue, String fromValue, String toValue, String bannerId, String event) {
+                                                 String attributeValue, String fromValue, String toValue, String bannerId, String event, List<BannerTriggerConditionEventProperty> eventProperties) {
     try {
-      if (events.isEmpty() || event == null || event.isEmpty()) {
+      if (events.isEmpty() || event == null || event.isEmpty() || compareValue == null || compareValue.isEmpty()) {
         return false;
       }
+
       for (TriggeredEvent triggeredEvent : events) {
-        if (triggeredEvent.getId() == null || !triggeredEvent.getId().equals(event) || compareValue == null || compareValue.isEmpty()) {
-          return false;
+        if (triggeredEvent.getId() == null || !triggeredEvent.getId().equals(event)) {
+          continue;
         }
-        List<TableBannerTrackEvent> bannerTrackEvents = DatabaseClient.getInstance(CleverPush.context)
+
+        if (eventProperties.size() > 0) {
+          if (!checkEventProperties(eventProperties, event)) {
+            return false;
+          }
+
+          for (BannerTriggerConditionEventProperty eventProperty : eventProperties) {
+            List<TableBannerTrackEvent> bannerTrackEvents = DatabaseClient.getInstance(CleverPush.context)
                 .getAppDatabase()
                 .trackEventDao()
-                .getBannerTrackEvent(bannerId, event, compareValue, attributeValue, String.valueOf(relation), fromValue, toValue);
+                .getBannerTrackEvent(bannerId, event, compareValue, attributeValue, String.valueOf(relation), fromValue, toValue, eventProperty.getRelation(), eventProperty.getProperty(), eventProperty.getValue());
+
+            if (bannerTrackEvents.isEmpty()) {
+              return false;
+            }
+
+            for (TableBannerTrackEvent bannerTrackEvent : bannerTrackEvents) {
+              int count = bannerTrackEvent.getCount();
+              String createdDate = bannerTrackEvent.getCreatedDateTime();
+              int pastDays = getDaysDifference(createdDate);
+
+              if (pastDays > Integer.parseInt(bannerTrackEvent.getProperty())) {
```

Comment:
whats bannerTrackEvent.getProperty()?

Reply:
bannerTrackEvent is list that we fetch from local DB.

<img width="424" alt="Screenshot 2024-06-11 at 4 00 14 PM" src="https://github.com/cleverpush/cleverpush-android-sdk/assets/129381063/fb0c2bde-f7be-43e4-a6b7-e4bf18af754f">

<img width="1373" alt="Screenshot 2024-06-11 at 3 51 00 PM" src="https://github.com/cleverpush/cleverpush-android-sdk/assets/129381063/568719af-2358-4cc0-85d1-06fd10404460">

Reply:
gotcha what happens for 0Days? and getDaysDifference(createdDate is same day?

Reply:
@cpKevinrodriguesnobre Can not add 0 in days.

Reply:
well perfect i guess